lib/irc.js

```diff
@@ -633,6 +140,16 @@ Client.prototype.chanData = function(name, create) {
     return this.chans[key];
 };
 
+Client.prototype._parseCmd = function(message) {
+    var self = this;
+    var fn = parseCmd[message.command];
+    if (fn != undefined) {
+        if (typeof(fn) == 'function') fn(self, message);
```

Comment:
Instead of passing `self`, we should call the functions in the context of `self`.